link/tracing.go

```diff
@@ -87,29 +88,72 @@ type TracingOptions struct {
 	// AttachTraceFEntry/AttachTraceFExit/AttachModifyReturn or
 	// AttachTraceRawTp.
 	Program *ebpf.Program
+	// Program attach type. Can be one of:
+	// 	- AttachTraceFEntry
+	// 	- AttachTraceFExit
+	// 	- AttachModifyReturn
+	// 	- AttachTraceRawTp
+	// If empty (AttachNone), the legacy attach method via raw tracepoint
+	// open will be used.
+	AttachType ebpf.AttachType
```

Comment:
We should probably carry this in an unexported field in Program instead. Can this field be retrieved from program info?